src/backend/logging/checkpoint/simple_checkpoint.h

```diff
@@ -31,30 +32,28 @@ class SimpleCheckpoint : public Checkpoint {
   SimpleCheckpoint &operator=(const SimpleCheckpoint &) = delete;
   SimpleCheckpoint(SimpleCheckpoint &&) = delete;
   SimpleCheckpoint &operator=(SimpleCheckpoint &&) = delete;
-  SimpleCheckpoint();
+  SimpleCheckpoint(bool test_mode);
   ~SimpleCheckpoint();
 
-  static SimpleCheckpoint &GetInstance();
-
   // Inherited functions
-  void Init();
-
   void DoCheckpoint();
 
   cid_t DoRecovery();
 
   // Internal functions
   void InsertTuple(cid_t commit_id);
 
-  bool Execute(executor::AbstractExecutor *scan_executor,
-               concurrency::Transaction *txn, storage::DataTable *target_table,
-               oid_t database_oid);
+  void Scan(storage::DataTable *target_table, oid_t database_oid);
 
   // Getters and Setters
   void SetLogger(BackendLogger *logger);
 
   std::vector<std::shared_ptr<LogRecord>> GetRecords();
```

Comment:
I'm not sure where this function gets used. Is it necessary to create a full copy of this vector every time?